src/common/parallel_for.cpp

```diff
@@ -22,17 +22,17 @@ namespace Qrack {
  * Iterate through the permutations a maximum of end-begin times, allowing the
  * caller to control the incrementation offset through 'inc'.
  */
-void ParallelFor::par_for_inc(const bitCapInt begin, const bitCapInt end, IncrementFunc inc, ParallelFunc fn)
+void ParallelFor::par_for_inc(const bitCapInt begin, const bitCapInt end, const bitCapInt itemCount, IncrementFunc inc, ParallelFunc fn)
```

Comment:
Seems like it might have been cleaner to remove the ‘end’ parameter entirely and replace it with just ‘count’ or ‘length’.

Reply:
"end" is not "begin + itemCount." Depending on the increment function, "itemCount" is potentially any number <= (end - begin). This is why I added the parameter.